pvlib/tracking.py

```diff
@@ -338,118 +373,84 @@ def singleaxis(apparent_zenith, apparent_azimuth,
+
+    # Calculate sun position (xp, yp, zp) in tracker coordinate system using
+    # [1] Eq 4.
+
+    cos_axis_azimuth = cosd(axis_azimuth)
+    sin_axis_azimuth = sind(axis_azimuth)
+    cos_axis_tilt = cosd(axis_tilt)
+    sin_axis_tilt = sind(axis_tilt)
+    xp = x*cos_axis_azimuth - y*sin_axis_azimuth
+    yp = (x*cos_axis_tilt*sin_axis_azimuth
+          + y*cos_axis_tilt*cos_axis_azimuth
+          - z*sin_axis_tilt)
+    zp = (x*sin_axis_tilt*sin_axis_azimuth
+          + y*sin_axis_tilt*cos_axis_azimuth
+          + z*cos_axis_tilt)
 
     # The ideal tracking angle wid is the rotation to place the sun position
     # vector (xp, yp, zp) in the (y, z) plane; i.e., normal to the panel and
     # containing the axis of rotation.  wid = 0 indicates that the panel is
-    # horizontal.  Here, our convention is that a clockwise rotation is
+    # horizontal. Here, our convention is that a clockwise rotation is
     # positive, to view rotation angles in the same frame of reference as
-    # azimuth.  For example, for a system with tracking axis oriented south,
-    # a rotation toward the east is negative, and a rotation to the west is
-    # positive.
-
-    # Use arctan2 and avoid the tmp corrections.
-
-    # angle from x-y plane to projection of sun vector onto x-z plane
-#     tmp = np.degrees(np.arctan(zp/xp))
-
-    # Obtain wid by translating tmp to convention for rotation angles.
-    # Have to account for which quadrant of the x-z plane in which the sun
-    # vector lies.  Complete solution here but probably not necessary to
-    # consider QIII and QIV.
-#     wid = pd.Series(index=times)
-#     wid[(xp>=0) & (zp>=0)] =  90 - tmp[(xp>=0) & (zp>=0)]  # QI
-#     wid[(xp<0)  & (zp>=0)] = -90 - tmp[(xp<0)  & (zp>=0)]  # QII
-#     wid[(xp<0)  & (zp<0)]  = -90 - tmp[(xp<0)  & (zp<0)]   # QIII
-#     wid[(xp>=0) & (zp<0)]  =  90 - tmp[(xp>=0) & (zp<0)]   # QIV
+    # azimuth. For example, for a system with tracking axis oriented south, a
+    # rotation toward the east is negative, and a rotation to the west is
+    # positive. This is a right-handed rotation around the tracker y-axis.
 
     # Calculate angle from x-y plane to projection of sun vector onto x-z plane
-    # and then obtain wid by translating tmp to convention for rotation angles.
-    wid = 90 - np.degrees(np.arctan2(zp, xp))
+    # using [1] Eq. 5.
+
+    wid = np.degrees(np.arctan2(xp, zp))
 
     # filter for sun above panel horizon
     zen_gt_90 = apparent_zenith > 90
     wid[zen_gt_90] = np.nan
 
-    # Account for backtracking; modified from [1] to account for rotation
-    # angle convention being used here.
+    # Account for backtracking
     if backtrack:
-        axes_distance = 1/gcr
-        # clip needed for low angles. GH 656
-        temp = np.clip(axes_distance*cosd(wid), -1, 1)
+        # distance between rows in terms of rack lengths relative to cross-axis
+        # tilt
+        axes_distance = 1/gcr/cosd(cross_axis_tilt)
```

Comment:
I would prefer `axes_distance = 1 / (gcr * cosd(cross_axis_tilt))  but that's a style preference